x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/filter_agg/components/filter_range_form.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback } from 'react';
+import {
+  EuiFieldNumber,
+  EuiFormRow,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiButtonToggle,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { FilterAggConfigRange } from '../types';
+
+/**
+ * Form component for the range filter aggregation for number type fields.
+ */
+export const FilterRangeForm: FilterAggConfigRange['aggTypeConfig']['FilterAggFormComponent'] = ({
+  config,
+  onChange,
+}) => {
+  const from = config?.from ?? '';
+  const to = config?.to ?? '';
+  const includeFrom = config?.includeFrom ?? false;
+  const includeTo = config?.includeTo ?? false;
+
+  const updateConfig = useCallback(
+    (update) => {
+      onChange({
+        config: {
+          ...config,
+          ...update,
+        },
+      });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [config]
+  );
+
+  return (
+    <>
+      <EuiSpacer size="m" />
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.transform.agg.popoverForm.filerAgg.range.greaterThanLabel"
+                defaultMessage="Greater than"
+              />
+            }
+          >
+            <EuiFieldNumber
+              value={from}
+              max={to !== '' ? to : undefined}
+              onChange={(e) => {
+                updateConfig({ from: e.target.value === '' ? undefined : Number(e.target.value) });
+              }}
+              step={0.1}
```

Comment:
I'm not sure it's worth setting this step to `0.1`. I still see red error bars under the `EuiFieldNumber` inputs depending on the values I enter. Even without the `step` it looks like you can still input e.g. `23.001`.